src/parser/priest/shadow/modules/core/Channeling.tsx

```diff
@@ -1,44 +1,73 @@
+    if (event.ability.guid === SPELLS.MIND_BLAST.id) {
+      this.mindBlastStarts[event.timestamp] = event; // Add to events list to check later
+    }
+    super.on_byPlayer_begincast(event);
+  }
+
   on_byPlayer_cast(event: CastEvent) {
     if (event.ability.guid === SPELLS.VOID_TORRENT_TALENT.id) {
       this.beginChannel(event);
       return;
     }
-    if (event.ability.guid === SPELLS.MIND_FLAY.id) {
+    if (event.ability.guid === SPELLS.MIND_FLAY.id || event.ability.guid === SPELLS.MIND_SEAR.id) {
       // Completely ignore this with regards to channeling since we use `applybuff` to track channel start, and this cast-event can occur as ticks too
       return;
     }
     super.on_byPlayer_cast(event);
   }
+
   // We can't use the `cast`-event for Mind Flay as this event can occur in the log during channel as ticks too
-  on_byPlayer_applybuff(event: ApplyBuffEvent) {
-    if (event.ability.guid === SPELLS.MIND_FLAY.id) {
+  on_byPlayer_applydebuff(event: ApplyDebuffEvent) {
+    if (event.ability.guid === SPELLS.MIND_FLAY.id || event.ability.guid === SPELLS.MIND_SEAR.id) {
       this.beginChannel(event);
     }
   }
 
-  cancelChannel(event: any, ability: any) {
-    if (this.isChannelingSpell(SPELLS.VOID_TORRENT_TALENT.id)) {
-      // If a channeling spell is "canceled" it was actually just ended, so if it looks canceled then instead just mark it as ended
-      this.log('Marking', this._currentChannel.ability.name, 'as ended since we started casting something else:', event.ability.name);
-      this.endChannel(event);
-    } else {
-      super.cancelChannel(event, ability);
+  // Looking at `removebuff` will includes progress towards a tick that never happened. This progress could be considered downtime as it accounts for nothing.
+  // If it's ever decided to consider the time between last tick and channel ending as downtime, just change the endchannel trigger.
+  on_byPlayer_removedebuff(event: RemoveDebuffEvent) {
+    if (CHANNEL_ABILITIES.every(ability => ability !== event.ability.guid)) {
+      return;
     }
+    if (!this.isChannelingSpell(SPELLS.VOID_TORRENT_TALENT.id) || !this.isChannelingSpell(SPELLS.MIND_FLAY.id) || !this.isChannelingSpell(SPELLS.MIND_SEAR.id)) {
+      return;
+    }
+    this.endChannel(event);
   }
 
-  // Looking at `removebuff` will includes progress towards a tick that never happened. This progress could be considered downtime as it accounts for nothing.
-  // If it's ever decided to consider the time between last tick and channel ending as downtime, just change the endchannel trigger.
-  on_byPlayer_removebuff(event: RemoveBuffEvent) {
-    if (event.ability.guid === SPELLS.MIND_FLAY.id || event.ability.guid === SPELLS.VOID_TORRENT_TALENT.id) {
+  cancelChannel(event: CastEvent, ability: Ability) {
+    if (CHANNEL_ABILITIES.some(ability => this.isChannelingSpell(ability))) {
+      if (event.ability.guid === SPELLS.MIND_BLAST.id) {
+        if (this.mindBlastStarts[event.timestamp]) {
```

Comment:
I saw that people check +- 100ms for instant casts, so this may not be the right way to do it? It was working for the events I tested with

Reply:
does the cast have a begincast event? Or is it just the cast event?

If it has a begincast as well, then you can do something like the below on the cast event
```
const variable = this.eventHistory.last(1, 100, Events.begincast.by(SELECTED_PLAYER).spell(SPELLS.WHATEVER_SPELL);
if (!variable.length) {
  //It was not instant
}
```
This will grab the last 1 spell within 100ms that matches the conditions (Begincast event that is a particular spell). If there isnt a begincast within 100ms (and therefore the variable is empty), then it wasnt instant. If there was a begincast found within 100ms, then it was instant.

That wont work though if the ability doesnt have a begincast event. For example, Fire Blast doesnt have a begincast event because it is always instant. But Pyroblast does have a begincast event because it is sometimes instant.

Reply:
It does have a begin cast event when it's instant, a little below I have an event listener for it and was adding to a list to check event timestamps.

Reply:
Updated it with the way similar to Pyroclasm. I like that a lot better, simpler and makes sense of what it's doing.